src/components/navigation/section-document-or-browser.tsx

```diff
@@ -196,20 +199,51 @@ export const SectionDocumentOrBrowser: React.FC<IProps> = ({ tabSpec, reset, sel
   };
 
   const showPlayback = user.isTeacher;
+  const [showPlaybackControls, setShowPlaybackControls] = useState(false);
+  const [documentToShow, setDocumentToShow] = useState<DocumentModelType>();
+
+  useEffect(() => {
+    setDocumentToShow(referenceDocument);
+  },[referenceDocument]);
+
+  const handleTogglePlaybackControlComponent = () => {
+    setShowPlaybackControls(!showPlaybackControls);
```

Comment:
It will be safer if you use a function in `setShowPlaybackControls`. This way it is guaranteed to work with the most recent version of the state value.
```suggestion
    setShowPlaybackControls(showControls => !showControls);
```